src/resolution/composite-arns-resolver.ts

```diff
@@ -80,20 +85,52 @@ export class CompositeArNSResolver implements NameResolver {
     }
 
     try {
-      const baseNameInCache =
-        await this.arnsNamesCache.getCachedArNSBaseName(baseName);
+      // start name list request before checking cache so that the name list
+      // stays fresh even when there are cache hits
+      const baseNameInCachePromise = this.arnsNamesCache
+        .getCachedArNSBaseName(baseName)
+        .catch((error: any) => {
+          this.log.error('Error getting base name from cache: ', {
+            message: error.message,
+            stack: error.stack,
+            baseName,
+          });
+          return undefined;
+        });
 
-      if (!baseNameInCache) {
-        this.log.warn('Base name not found in ArNS names cache', { name });
-        metrics.arnsNameCacheMissCounter.inc();
-        return {
-          name,
-          resolvedId: undefined,
-          resolvedAt: undefined,
-          ttl: undefined,
-          processId: undefined,
-        };
-      }
+      // Make a resolution function that can be called both on cache hits (when
+      // the refresh interval is past) and when we have a cache miss
+      const resolveName = async () => {
+        for (const resolver of this.resolvers) {
+          try {
+            this.log.info('Attempting to resolve name with resolver', {
+              type: resolver.constructor.name,
```

Comment:
The class is already added to the logger in the constructor

Reply:
Fixed a2202a47e1f1262654fec9b92f9dfa1bf522d651